tools/distpartitioning/data_shuffle.py

```diff
@@ -473,12 +472,19 @@ def exchange_feature(
         )
 
     # exchange actual data here.
-    if featdata_key != None:
+    logging.debug(f"Rank: {rank} {featdata_key.shape=}")
+    if featdata_key is not None:
         feat_dims_dtype = list(featdata_key.shape)
+        # When a feature is 2-dim, the shape should match the feature dimension.
+        if len(featdata_key.shape) == 2:
```

Comment:
add an assertion on it's `1` or `2`? If anyone in the future passes other dim feature, we could capture the error more explicitly.